source/common/upstream/load_balancer_impl.cc

```diff
@@ -487,84 +472,43 @@ void RoundRobinLoadBalancer::refresh(uint32_t priority) {
   }
 }
 
-HostConstSharedPtr RoundRobinLoadBalancer::chooseHost(LoadBalancerContext*) {
+HostConstSharedPtr EdfLoadBalancerBase::chooseHost(LoadBalancerContext*) {
   const HostsSource hosts_source = hostSourceToUse();
   auto scheduler_it = scheduler_.find(hosts_source);
   // We should always have a scheduler for any return value from
   // hostSourceToUse() via the construction in refresh();
   ASSERT(scheduler_it != scheduler_.end());
   auto& scheduler = scheduler_it->second;
-  if (scheduler.weighted_) {
+
+  // As has been commented in both EdfLoadBalancerBase::refresh and
+  // BaseDynamicClusterImpl::updateDynamicHostList, we must do a runtime pivot here to determine
+  // whether to use EDF or do unweighted (fast) selection.
+  // TODO(mattklein123): As commented elsewhere, this is wasteful, and we should just refresh the
+  // host set if any weights change.
+  if (stats_.max_host_weight_.value() != 1) {
```

Comment:
Are we worried in the P2C case that if we have identical weights that are not 1, e.g. all hosts are weighted at 42, we'll get subtley different behavior due to the EDF approach than if they are all 1?

Reply:
TBH not really. I think that's a pretty extreme edge case, but I will drop a comment in.